src/main.rs

```diff
@@ -1,3 +1,204 @@
-fn main() {
-    println!("Hello, world!");
+use dashmap::DashMap;
+use ropey::Rope;
+use tower_lsp::{
+    jsonrpc::Result,
+    lsp_types::{
+        Diagnostic, DiagnosticSeverity, DidChangeTextDocumentParams, DidOpenTextDocumentParams,
+        InitializeParams, InitializeResult, InitializedParams, NumberOrString, Position, Range,
+        ServerCapabilities, TextDocumentSyncCapability, TextDocumentSyncKind, Url,
+    },
+    Client, LanguageServer, LspService, Server,
+};
+use tree_sitter_lint::{
+    tree_sitter::{self, InputEdit, Parser, Point, Tree},
+    tree_sitter_grep::{Parseable, SupportedLanguage},
+};
+
+#[derive(Debug)]
+struct Backend {
+    client: Client,
+    per_file: DashMap<Url, PerFileState>,
+}
+
+impl Backend {
+    pub fn new(client: Client) -> Self {
+        Self {
+            client,
+            per_file: Default::default(),
+        }
+    }
+
+    async fn run_linting_and_report_diagnostics(&self, uri: &Url) {
+        let per_file_state = self.per_file.get(uri).unwrap();
+        let violations = tree_sitter_lint_local::run_for_slice(
+            &per_file_state.contents,
+            Some(&per_file_state.tree),
+            "dummy_path",
+            Default::default(),
+        );
+        self.client
+            .publish_diagnostics(
+                uri.clone(),
+                violations
+                    .into_iter()
+                    .map(|violation| Diagnostic {
+                        message: violation.message,
+                        range: tree_sitter_range_to_lsp_range(
+                            &per_file_state.contents,
+                            violation.range,
+                        ),
+                        severity: Some(DiagnosticSeverity::ERROR),
+                        code: Some(NumberOrString::String(violation.rule.name)),
```

Comment:
This mimics what I saw that the ESLint LSP server (from however I have that set up) is doing, which results in showing the rule name "as part of" the diagnostic